### ISSUE TITLE
Latest docker hub container looks broken

### ISSUE THREAD

**Issue by Chanonry** [localstack/localstack#3756]
This is a ...

x bug report

# Detailed description

Latest container was pushed 6 hours ago.

My local tests against a docker-compose container work. When running a GitHub action on a Pull Request the latest container was downloaded and I started getting 500  type errors

Reverted to 0.12.7 and everything is ok

## Expected behavior

Should be the same behaviour

## Actual behavior

Variable. Timed out connection on one test. Failed to delete s3 bucket on another

# Steps to reproduce

Dunno

## Command used to start LocalStack

docker- compose up

**Comment by mgagliardo** [comment]
Hello @Chanonry, there has been an issue with the DNS that back our localstack service, would you please check this again? You can also check the #community-support channel on slack. Also it would help (if this keeps happening) to have some examples or debug logs to reproduce this.

Thanks!

**Comment by mgagliardo** [comment]
Some local testing I've done:

```
$ awslocal s3api create-bucket --bucket test
$ awslocal s3api list-buckets
{
    "Buckets": [
        {
            "Name": "test",
            "CreationDate": "2021-03-19T17:00:14+00:00"
        }
    ],
    "Owner": {
        "DisplayName": "webfile",
        "ID": "bcaf1ffd86f41161ca5fb16fd081034f"
    }
}
```

```
$ awslocal s3api delete-bucket --bucket test
$ awslocal s3api list-buckets
{
    "Buckets": [],
    "Owner": {
        "DisplayName": "webfile",
        "ID": "bcaf1ffd86f41161ca5fb16fd081034f"
    }
}
```

**Comment by Chanonry** [comment]
I was using latest.

Its not really an issue for me as I just reverted to an image which was more in line with what I was using locally. 

Having said that I re ran my local tests forcing a docker-compose rebuild with both 0.12.7 and latest. Both were successful.

The errors I got were in a GitHub Action on the ubuntu-20.04 runner:

Read timeout on endpoint URL: "http://localhost:4566/"

and separately

 main_test.go:203: 	✗✗✗	 failed to delete bucket: nk-test-bucket-pdf-main-worker err: exceeded maximum number of attempts, 3, InternalServerError: Internal Server Error
151
            	status code: 500, request id: , host id:

**Comment by mgagliardo** [comment]
Ok so this is related to github actions, not localstack itself. This is probably because the runner runs localstack on another separated container and tries to connect to `localhost` which actually has nothing running (which is fine). Please try using `HOSTNAME_EXTERNAL` to test more.

This line clearly shows its a connection issue, more than a localstack issue.
`Read timeout on endpoint URL: "http://localhost:4566/"`

**Comment by Chanonry** [comment]
If that is the case then why does 0.12.7 run then?  The problem manifests itself  your latest container.

Think you are a bit quick to say its not your issue.

Not mine either, you can debug your own product

**Comment by mgagliardo** [comment]
Ok maybe I missunderstood the problem, my mistake. I thought you tested locally with both `latest` and `0.12.7` and as it was working then it was fine.

Can you please share your GitHub action pipeline config?

**Comment by at-wat** [comment]
I'm seeing the same problem.
Latest localstack returns 500 error when calling s3 API and it is worked around by reverting localstack version to 0.12.7. (https://github.com/seqsense/s3sync/pull/133)

Steps to reproduce (requires Golang):
```shell
$ docker pull localstack/localstack
$ git clone https://github.com/seqsense/s3sync.git -b v1.8.0
$ cd s3sync
$ make s3 &

# wait localstack initialization complete

$ make fixture
$ go test . -v -count=1
```

You'll see errors like:
```
=== RUN   TestS3sync
=== RUN   TestS3sync/Download
2021/03/22 12:52:32 Downloading bar/baz/README.md to /tmp/s3synctest752909010/bar/baz/README.md
2021/03/22 12:52:32 Downloading README.md to /tmp/s3synctest752909010/README.md
2021/03/22 12:52:32 Downloading foo/README.md to /tmp/s3synctest752909010/foo/README.md
2021-03-22T03:52:32:WARNING:bootstrap.py: Thread run method <function AdaptiveThreadPool.submit.<locals>._run at 0x7f9dd6bf2c10>(None) failed: An error occurred (404) when calling the HeadObject operation: Not Found Traceback (most recent call last):
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 671, in run
    result = self.func(self.params)
  File "/opt/code/localstack/localstack/utils/async_utils.py", line 28, in _run
    return fn(*args, **kwargs)
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 607, in handler
    response = modify_and_forward(method=method, path=path_with_params, data_bytes=data, headers=headers,
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 373, in modify_and_forward
    listener_result = listener.forward_request(method=method,
  File "/opt/code/localstack/localstack/services/edge.py", line 108, in forward_request
    return do_forward_request(api, method, path, data, headers, port=port)
  File "/opt/code/localstack/localstack/services/edge.py", line 119, in do_forward_request
    result = do_forward_request_inmem(api, method, path, data, headers, port=port)
  File "/opt/code/localstack/localstack/services/edge.py", line 139, in do_forward_request_inmem
    response = modify_and_forward(method=method, path=path, data_bytes=data, headers=headers,
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 443, in modify_and_forward
    updated_response = update_listener.return_response(**kwargs)
  File "/opt/code/localstack/localstack/services/s3/s3_listener.py", line 1253, in return_response
    fix_range_content_type(bucket_name, path, headers, response)
  File "/opt/code/localstack/localstack/services/s3/s3_listener.py", line 504, in fix_range_content_type
    result = s3_client.head_object(Bucket=bucket_name, Key=key_name)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 676, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (404) when calling the HeadObject operation: Not Found

2021-03-22T03:52:32:WARNING:localstack.utils.server.http2_server: Error in proxy handler for request GET http://localhost:4572/example-bucket/README.md: An error occurred (404) when calling the HeadObject operation: Not Found Traceback (most recent call last):
  File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 129, in index
    raise result
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 671, in run
    result = self.func(self.params)
  File "/opt/code/localstack/localstack/utils/async_utils.py", line 28, in _run
    return fn(*args, **kwargs)
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 607, in handler
    response = modify_and_forward(method=method, path=path_with_params, data_bytes=data, headers=headers,
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 373, in modify_and_forward
    listener_result = listener.forward_request(method=method,
  File "/opt/code/localstack/localstack/services/edge.py", line 108, in forward_request
    return do_forward_request(api, method, path, data, headers, port=port)
  File "/opt/code/localstack/localstack/services/edge.py", line 119, in do_forward_request
    result = do_forward_request_inmem(api, method, path, data, headers, port=port)
  File "/opt/code/localstack/localstack/services/edge.py", line 139, in do_forward_request_inmem
    response = modify_and_forward(method=method, path=path, data_bytes=data, headers=headers,
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 443, in modify_and_forward
    updated_response = update_listener.return_response(**kwargs)
  File "/opt/code/localstack/localstack/services/s3/s3_listener.py", line 1253, in return_response
    fix_range_content_type(bucket_name, path, headers, response)
  File "/opt/code/localstack/localstack/services/s3/s3_listener.py", line 504, in fix_range_content_type
    result = s3_client.head_object(Bucket=bucket_name, Key=key_name)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 676, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (404) when calling the HeadObject operation: Not Found

    s3sync_test.go:76: Sync should be successful InternalServerError: Internal Server Error
        	status code: 500, request id: , host id: 
```

**Comment by ptdel** [comment]
not sure if this is related, but I'm able to create s3 bucket objects and get a return from localstack but trying to create a bucket (`aws s3api create-bucket --bucket test-bucket`) doesn't return any output and subsequent tests trying to use the bucket fail with a 404

**Comment by macnev2013** [comment]
Thanks @at-wat for providing the repository. It will help us a lot.

**Comment by whummer** [comment]
Thanks @at-wat @Chanonry @macnev2013 . This should be fixed in #3800 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks!

**Comment by at-wat** [comment]
Thank you! I could confirm that our test works with the latest localstack image!
